routers/web/repo/pull.go

```diff
@@ -1411,3 +1412,31 @@ func UpdatePullRequestTarget(ctx *context.Context) {
 		"base_branch": pr.BaseBranch,
 	})
 }
+
+// SetAllowEdits allow edits from maintainers to PRs
+func SetAllowEdits(ctx *context.Context) {
+	form := web.GetForm(ctx).(*forms.UpdateAllowEditsForm)
+
+	pr, err := models.GetPullRequestByIndex(ctx.Repo.Repository.ID, ctx.ParamsInt64(":index"))
+	if err != nil {
+		if models.IsErrPullRequestNotExist(err) {
+			ctx.NotFound("GetPullRequestByIndex", err)
+		} else {
+			ctx.ServerError("GetPullRequestByIndex", err)
+		}
+		return
+	}
+
+	if err := pull_service.SetAllowEdits(ctx, ctx.Doer, pr, form.AllowMaintainerEdit); err != nil {
```

Comment:
TODO: form.AllowMaintainerEdit is always false - please fix this

Reply:
IIRC if you want to post JSON to backend, you need `data: JSON.stringify({_csrf: csrfToken, allowMaintainerEdit: checked})` in JS code, and maybe add `application/json` header (sometimes not necessary, it depends on the backend fraemwork).

Reply:
sorry, misread the code. you are using form not json 😂

it might be caused some other problems that I haven't found ....

Reply:
Fixed, the item name in the form requires snake_case.